server/src/main/kotlin/org/jetbrains/bsp/bazel/server/sync/ClasspathQuery.kt

```diff
@@ -15,8 +16,17 @@ object ClasspathQuery {
                 .executeBazelCommand(parseProcessOutput = false)
                 .waitAndGetResult(cancelChecker, ensureAllOutputRead = true)
         if (cqueryResult.isNotSuccess) throw RuntimeException("Could not query target '${target.uri}' for runtime classpath")
-        val classpaths = Gson().fromJson(cqueryResult.stdout, JvmClasspath::class.java)
-        return classpaths
+        try {
+            val classpaths = Gson().fromJson(cqueryResult.stdout, JvmClasspath::class.java)
+            return classpaths
+        } catch (e: JsonSyntaxException){
+            // sometimes Bazel returns two values to a query when multiple configurations apply to a target
+            return if (cqueryResult.stdoutLines.size > 1) {
+                val allOpts = cqueryResult.stdoutLines.map { Gson().fromJson(it, JvmClasspath::class.java) }
+                allOpts.maxByOrNull { it.runtime_classpath.size + it.compile_classpath.size }!!
```

Comment:
Changed the logic to find the longest classpath, but as I said it's just a heuristic